spec/kracken/authenticator_spec.rb

```diff
@@ -0,0 +1,62 @@
+require 'spec_helper'
+
+module Kracken
+  describe Authenticator do
+
+
+    describe ".with_credentials" do
+      it "returns nil when nothing is found" do
+        expect_any_instance_of(CredentialAuthenticator)
+          .to receive(:fetch)
+          .and_return(nil)
+
+        expect(Authenticator.user_with_credentials("melody@ponds.uk", "secret")).to be_nil
+      end
+
+      it "creates a user using the user_class" do
+        expect_any_instance_of(CredentialAuthenticator)
+          .to receive(:fetch)
+          .and_return({'attributes' => {'uid' => 1}})
+
+        expect(Authenticator.user_with_credentials("melody@ponds.uk", "secret").class).to eq User
```

Comment:
Seems a little magical. The description uses `user_class` but the expectation uses `User`.